python/pyspark/context.py

```diff
@@ -21,6 +21,8 @@
 from threading import Lock
 from tempfile import NamedTemporaryFile
 
+from pip.commands.install import InstallCommand as pip_InstallCommand
```

Comment:
This introduce a dependency on `pip`, which is not available by default. We should put it in a `try`.